### ISSUE TITLE
github will replace and discontinue authenticating using query parameters

### ISSUE THREAD

**Issue by yb199478** [composer/composer#8454]
I recently read the oauth documentation on github and found that the mention of deprecating authentication to the GitHub API using query parameters:

- https://developer.github.com/changes/2019-11-05-deprecated-passwords-and-authorizations-api/#authenticating-using-query-parameters

Will this affect composer in the future?

https://github.com/composer/composer/blob/a0690525b52f841748176e1c43dbe900f07e72da/src/Composer/Util/RemoteFilesystem.php#L281

**Comment by alcohol** [comment]
A quick glance over the blogpost and our code suggests that this might impact some of our code perhaps. But I was also under the impression we already used headers to authenticate in some scenarios or flows. This does warrant further inspection I think. Thank you for letting us know!

**Comment by stof** [comment]
The blog post also mention deprecating the API for creating personal access tokens. I know for sure that composer used it in the past, but I don't remember whether that's still the case today.

**Comment by yb199478** [comment]
The API for creating the personal token, I remember that it provided a uri to display, not directly generated through the api:

https://github.com/composer/composer/blob/e8626505718981bf67fd7262f65a611d4000f6c6/src/Composer/Util/GitHub.php#L89

**Comment by stof** [comment]
OK, so it means we already removed our code generating a token.

**Comment by Rarst** [comment]
FYI GitHub have started proactively notifying about this, I just got the following email:

> On February 4th, 2020 at 07:59 (UTC) your personal access token ( ... ) using Composer/1.9.1 (Windows NT; 10.0; PHP 7.4.1) was used as part of a query parameter to access an endpoint through the GitHub API:
> ...
> Please use the Authorization HTTP header instead, as using the `access_token` query parameter is deprecated.

Since the fix doesn't seem to be in stable release yet, this might go out a lot. :)

**Comment by Seldaek** [comment]
Yup, saw this too. Will definitely get to get 1.10 out next week with this fix.

**Comment by glensc** [comment]
They'll keep sending that email out every 3 days:

> Depending on your API usage, we'll be sending you this email reminder once every 3 days for each token and User-Agent used in API calls made on your behalf.
Just one URL that was accessed with a token and User-Agent combination will be listed in the email reminder, not all.

**Comment by Seldaek** [comment]
1.9.3 is now out with the fix. `composer self-update` party.

**Comment by taisph** [comment]
I'm still getting the GitHub notification with composer 1.9.3. :confused: 

```
On February 14th, 2020 at 14:39 (UTC) your personal access token (Cloud Build) using Composer/1.9.3 (Linux; 4.15.0-1044-gcp; PHP 5.6.40) was used as part of a query parameter to access an endpoint through the GitHub API:
```

**Comment by Seldaek** [comment]
@taisph make sure you remove hirak/prestissimo if you have the plugin installed, at least if you care about stopping the flow of emails.

**Comment by taisph** [comment]
@Seldaek Ah. I didn't even consider that possibility. Thank you for pointing me in that direction. 

Seems like it has been fixed with https://github.com/hirak/prestissimo/pull/211 and released in hirak/prestissimo 0.3.10.

**Comment by WinstonN** [comment]
> 1.9.3 is now out with the fix. `composer self-update` party.

@Seldaek Thank you for being the coolest guy ever. I met you in Cape Town, at RunwaySale when you gave your Regex 101 talk, and really enjoyed meeting you! Hope you're doing well! <3
